ledger/caching/src/test/suite/scala/com/daml/caching/SizedCacheSpec.scala

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.caching
+
+import scala.util.Random
+
+class SizedCacheSpec extends CacheSpecBase("a sized cache") {
+  override protected def newCache(): Cache[Integer, String] =
+    SizedCache.from[Integer, String](SizedCache.Configuration(maximumSize = 16))
+
+  "a sized cache" should {
+    "evict values eventually, once the size limit has been reached" in {
+      val cache =
+        SizedCache.from[Integer, String](SizedCache.Configuration(maximumSize = 256))
+      val values = Iterator.continually[Integer](Random.nextInt).take(1000).toSet.toVector
+
+      values.foreach { value =>
+        cache.get(value, _.toString)
+      }
+      val cachedValues = values.map(cache.getIfPresent).filter(_.isDefined)
+
+      cachedValues.length should (be > 16 and be < 500)
```

Comment:
Because eviction might be delayed; Caffeine has some heuristics about when to evict that I don't quite understand. As you say, it's not a hard limit.

It also might evict _more_ than you expect, because again, fun times.

Reply:
That's fine as long as the test case won't be flaky due to this variability (and the expected length range).

Reply:
I ran them a hundred times to be sure. 🙂

Reply:
Good call!